sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -364,9 +367,21 @@ private Mono<RxDocumentServiceResponse> toDocumentServiceResponse(Mono<HttpRespo
                     // If there is any error in the header response this throws exception
                     validateOrThrow(request, HttpResponseStatus.valueOf(httpResponseStatus), httpResponseHeaders, content);
 
-                    StoreResponse rsp = new StoreResponse(httpResponseStatus,
-                        HttpUtils.unescape(httpResponseHeaders.toMap()),
-                        content);
+                    StoreResponse rsp;
+
+                    if (content != null && content.length > 0) {
+                        final ByteBuf buffer = Unpooled.wrappedBuffer(content);
```

Comment:
This conversion of byte array to ByteBuf is doing double-deserialization of response body. If you notice the `response.bodyAsByteArray()`, it converts response body from `ByteBuf` to `byte[]`. In my opinion, a better way would be to use response.body() - which returns Flux<ByteBuf> directly on line 352 - 355

```
Flux<ByteBuf> body = httpResponse.body().switchIfEmpty(Flux.just(Unpooled.EMPTY_BUFFER));
```
And then we don't need to do this extra conversion here.

Reply:
> Currently we are just using byte array from the response body, but with this change, with every response, we will be creating an additional byte buffer wrapping the content byte array on heap. Will this have extra memory effects?

I will allocate a single pointer - only in Gateway communication - vs. a pointer for a new byte[] plus the actual byte[] on direct hot path. So, I am not too concerned about the side effect here.

Reply:
> This conversion of byte array to ByteBuf is doing double-deserialization of response body. If you notice the `response.bodyAsByteArray()`, it converts response body from `ByteBuf` to `byte[]`. In my opinion, a better way would be to use response.body() - which returns Flux directly on line 352 - 355
> 
> ```
> Flux<ByteBuf> body = httpResponse.body().switchIfEmpty(Flux.just(Unpooled.EMPTY_BUFFER));
> ```
> 
> And then we don't need to do this extra conversion here.

We would then still need to convert the Flux<ByteBuf> back into a single ByteBuf - reactor.netty.ByteBufFlex has this functionality in its aggregate method - but we can't rely on HttpResponse.body always being a ByteBufFlux our own BufferedHttpResponse for example would break that assumption.

So, I would need to also change HttpResponse.body to require ByteBufFlux - Doable, but my preference would be to do that in a separate PR. Any concerns with that?

Reply:
Actually simpler than I thought. I added the refactoring - making body Mono<ByteBuf> (with the aggregated ByteBuf) - only perf overhead would be in BufferedHttpresponse which is only ever used in tests anyway